src/Compilers/VisualBasic/Test/Syntax/Parser/ParseMethods.vb

```diff
@@ -346,8 +346,6 @@ Public Class ParseMethods
             End Class
         ]]>,
         <errors>
-            <error id="30201"/>
```

Comment:
Yes. Either copy the test to a Sub Bug862505_LangVersion15 or keep it all in one method and do the existing test under the default options, then change the parse options and reparse and assert the new behavior. Using the For loop is a bit round about for what's really just two tests. No nesting or conditional logic required. Like this:

    Public Sub Bug862505()
        Dim code =
"Class C1
    Function f1(Optional ByVal c1 As New Object())
    End Function
End Class"

        ParserTestUtilities.ParseAndVerify(code,
                                           VisualBasicParseOptions.Default,
                                           <errors>
                                               <error id="30201"/>
                                               <error id="30812"/>
                                               <error id="30180"/>
                                           </errors>)

        ' In VB '15' you can omit the default value so only BC30180 applies.
        ParserTestUtilities.ParseAndVerify(code,
                                           VisualBasicParseOptions.Default.WithLanguageVersion(LanguageVersion.VNext),
                                           <errors>
                                               <error id="30180"/>
                                           </errors>)
    End Sub



---
In reply to: [63098720](https://github.com/dotnet/roslyn/pull/11180#discussion_r63098720) [](ancestors = 63098720)